### ISSUE TITLE
std.net : Add doc comment for tcpConnectToAddress mentioning the format of address

### ISSUE THREAD

**Issue by palash25** [ziglang/zig#5471]
Hi, new to zig, started learning a few days ago, was playing around with it trying to connect to tcp server and I kept getting an error for my code which looked roughly like this

```zig
    var port: u16 = 6379;
    var addr = "localhost";
    var conn = try std.net.tcpConnectToAddress(try net.Address.parseIp4(addr, port));
```

the error said `InvalidCharacter` so I had to do a little digging and saw that while parsing the address the code was looking for "." which gave me the clue to use `127.0.0.1` instead of `localhost` and my code worked.

The TCP libraries that I have played around with in other languages accept both `127.0.0.1` and `localhost` but since it is not the case with Zig I was wondering if we should mention this at the top of the `tcpConnectToAddress` function so that the users know right off the bat what kind of address to supply the function with.

Let me know if you would like a PR for this. :smile:

**Comment by SuperAuguste** [comment]
Hey @palash25, welcome to the Zig community 🎉!

I've read your issue and disagree with you about the ambiguity of the `tcpConnectToAddress` function, because the `parseIp4` function performs the parsing of the host, and in my opinion, it is adequately named as it only accepts IPv4 formatted addresses.

I think a topic that could definitely be up for discussion is potentially resolving `localhost` to `127.0.0.1` in `parseIp4`, though.

**Comment by palash25** [comment]
you are right @SuperAuguste I didn't realise that we are already passing `parseIp4` to the function, that was ignorant of me, thanks for pointing it out. :sweat_smile: 

I will change the title and description to add a special localhost case in `parseIp4` function

**Comment by daurnimator** [comment]
> I think a topic that could definitely be up for discussion though is potentially resolving `localhost` to `127.0.0.1` in `parseIp4`.

I don't like the idea of that.

**Comment by iansimonson** [comment]
I think you can use `tcpConnectToHost` instead which I believe works with both IP and hostname (did a quick test to confirm)

**Comment by andrewrk** [comment]
"parseIP" functions should not parse non-IP addresses. Seems like OP simply needs to call tcpConnectToHost.
